File: examples/js/embeddings-sample.js
import { OpenAI } from 'openai';
import fs from 'fs';


const client = new OpenAI({
  apiKey: 'dpais',
  baseURL: 'http://localhost:8553/v1/openai',
});

// Text Embedding Request
const textResponse = await client.embeddings.create({
  model: 'clip',
  input: ['otter', 'horse'],
  encoding_format: 'float',
});

// Convert Image to Base64
const imagePath = 'path/to/imageofanotter.jpg';
const imageBuffer = fs.readFileSync(imagePath);
const encodedImage = imageBuffer.toString('base64');

// Image Embedding Request
const imageResponse = await client.embeddings.create({
  model: 'openai-clip',
  encoding_format: 'float',
  images: [encodedImage]  
  }
console.log(imageResponse);

File: examples/python/embeddings-sample.py
# Please install OpenAI SDK first: `pip3 install openai`


from openai import OpenAI
import base64
client = OpenAI(api_key="dpais", base_url="http://localhost:8553/v1/openai")

textResponse = client.embeddings.create(
  model="openai-clip",
  input="The food was delicious and the waiter...",
  encoding_format="float"
)

with open("path/to/image.jpg", "rb") as image_file:
  encoded_image = base64.b64encode(image_file.read()).decode("utf-8")

image_response = client.embeddings.create(
  model="openai-clip",
  input=None,  
  encoding_format="float",
  extra_body={"images": [encoded_image] } # Pass images as base64 string
)